erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -253,17 +253,24 @@ def get_directions(self, route, optimize):
 
 
 @frappe.whitelist()
-def get_delivery_window(customer=None):
-	delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
-	delivery_window.default_window = False
-
-	if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+def get_delivery_window(doctype=None, docname=None, customer=None):
+	if customer:
+		delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
+		delivery_window.default_window = False
+
+		if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+			return delivery_window
+		
+		default_window = frappe.db.get_value("Delivery Settings", "Delivery Settings", ["delivery_start_time", "delivery_end_time"], as_dict=1)
+		if default_window.delivery_start_time and default_window.delivery_end_time:
+			default_window.default_window = True
+			return default_window
+		else:
+			frappe.throw(_("Default Delivery Window not available"))
+	elif doctype and docname:
```

Comment:
Change the priority order to "Transaction" > "Customer" > "Delivery Settings". Move this check before the customer check.